File: leetcode/majority-element.go
/*
- Problem: https://leetcode.com/problems/majority-element/
*/

func majorityElement(nums []int) int {
    var trackMap map[int]int = map[int]int{}
    var nlen int = len(nums)
    var maxV int = 0
    var maxK int = 0
    
    for _, n := range nums {
        if _, ok := trackMap[n]; ok {
            trackMap[n] = trackMap[n] + 1
        } else {
            trackMap[n] = 1
        }
    }
    
    for k, v := range trackMap {
        if v >= nlen / 2 {
            if maxV < v {
                maxV = v
                maxK = k
            }
        }
    }
    
    return maxK
}

File: leetcode/java/majority-element.java
/*
- Problem: https://leetcode.com/problems/majority-element/
*/

class Solution {
    public int majorityElement(int[] nums) {
        
        HashMap<Integer, Integer> countMap = new HashMap<Integer, Integer>();
        int maxCount = 1;
        int maxN = nums[0];

        for (int n : nums) {
            if (countMap.containsKey(n)) {
                countMap.put(n, countMap.get(n)+1);
            } else {
                countMap.put(n, 1);
            }
            if (maxCount < countMap.get(n)) {
                maxCount = countMap.get(n);
                maxN = n;
            }
        }

        return maxN;
        
    }
}
